### PR TITLE
Add contribution guidelines.

### PR DESCRIPTION
**Pull request by berndbohmeier**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Please also include the version of the relay you were using, your operating system and the used configuration.
+
+### Request a feature
+Please include a description of what you would like to see implemented and an explanation of why you believe this would
+ be a good addition.
+
+## Open a pull request
+
+### Before starting
+- Ensure that you can run the tests and that they pass. The tests can be run with from the root with
```
Comment (cducrest) [inline CONTRIBUTING.md:27]:
```suggestion
- Ensure that you can run the tests and that they pass. The tests can be run from the root with
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+- Ensure that you can run the end2end tests. You can find them in the
+[end2end repo](https://github.com/trustlines-protocol/end2end):
+  First build the docker image locally from the root of this repo with
+  `docker build -t <tag-name> .`
+  Then follow the instructions of the end2end repository to run the tests with that image.
+
+### To include
+
+- For a bug fix:
+  - Add a test that shows that the bug was fixed, and make sure that the test fails
```
Comment (cducrest) [inline CONTRIBUTING.md:38]:
```suggestion
  - Add a test that makes the bug explicit, and make sure that the test fails
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  - Fix the bug, so that the test passes
+  - For bugs, you should consider adding an end2end test
+- For a new feature:
+  - Add a test, that demonstrate the new feature. Consider also adding an end2end test to the
+    [end2end repo](https://github.com/trustlines-protocol/end2end) if necessary.
```
Comment (cducrest) [inline CONTRIBUTING.md:43]: Isn't it confusing that we ask them to add tests to the e2e repo while they live in the clientlib?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  - Add or update related documentation if necessary.
+- Add an entry to the unreleased section of the [changelog](CHANGELOG.rst).
+- Format your code changes with black. If you setup pre-commit, this will happen automatically on commit
+  `black setup.py tests src`
+- Check your code with `flake8` and `mypy`. If you setup pre-commit, this will happen automatically on commit.
```
Comment (cducrest) [inline CONTRIBUTING.md:48]: Maybe explain how to set up pre-commit

File: CONTRIBUTING.md (hunk 3 above)
Comment (berndbohmeier) [reply CONTRIBUTING.md:43]: ah, mmh yes, well I guess they should actually live there :D 
I'll change it.

File: CONTRIBUTING.md (hunk 4 above)
Comment (berndbohmeier) [reply CONTRIBUTING.md:49]: I pointed out to follow the readme, there it is documented.

**Pull request merged**